treeherder/webapp/api/performance_serializers.py

```diff
@@ -385,6 +385,140 @@ def get_name(self, value):
         return '{} {} {}'.format(test_suite, value['option_name'], value['extra_options'])
 
 
+class PerfCompareResultsQueryParamsSerializer(serializers.Serializer):
+    base_revision = serializers.CharField(required=False, allow_null=True, default=None)
+    new_revision = serializers.CharField(required=False, allow_null=True, default=None)
+    base_repository = serializers.CharField()
+    new_repository = serializers.CharField()
+    framework = serializers.IntegerField(required=False, allow_null=True, default=None)
+    interval = serializers.IntegerField(required=False, allow_null=True, default=None)
+    no_subtests = serializers.BooleanField(required=False)
+
+    def validate(self, data):
+        if (
+            data['base_revision'] is None
+            and data['new_revision'] is None
+            and data['interval'] is None
+        ):
+            raise serializers.ValidationError('Required: base_revision, new_revision and interval.')
+
+        return data
+
+    def validate_base_repository(self, base_repository):
+        try:
+            Repository.objects.get(name=base_repository)
+
+        except ObjectDoesNotExist:
+            raise serializers.ValidationError('{} does not exist.'.format(base_repository))
+
+        return base_repository
+
+    def validate_new_repository(self, new_repository):
```

Comment:
can't we have one method `validate_repository` as those 2 are doing the same thing?